pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenvNoCC, lib, writeTextDir, php, makeBinaryWrapper }:
+
+let
+  buildPhpProjectOverride = finalAttrs: previousAttrs:
+
+  let
+    phpDrv = finalAttrs.php or php;
+    composer = finalAttrs.composer or phpDrv.packages.composer;
+    composerLock = finalAttrs.composerLock or null;
+  in {
+    nativeBuildInputs = (previousAttrs.nativeBuildInputs or []) ++ [
+      composer
+      phpDrv.composerHooks.composerInstallHook
+    ];
+
+    buildInputs = (previousAttrs.buildInputs or []) ++ [
+      phpDrv
+    ];
+
+    strictDeps = true;
+
+    # Should we keep these empty phases?
+    configurePhase = (previousAttrs.configurePhase or "") + ''
+      runHook preConfigure
+
+      runHook postConfigure
+    '';
+
+    buildPhase = (previousAttrs.buildPhase or "") + ''
+      runHook preBuild
+
+      runHook postBuild
+    '';
+
+    installPhase = (previousAttrs.installPhase or "") + ''
+      runHook preInstall
+
+      runHook postInstall
+    '';
```

Comment:
Here you do not want to merge these (otherwise you would potentially run the hooks multiple times) but use the empty phases as the `or` fallback.

Reply:
```nix
installPhase = (previousAttrs.installPhase or ''
  runHook preInstall
  runHook postInstall
'');
```  
?

Reply:
Fixed.